solidity/contracts/system/KeepFactorySelection.sol

```diff
@@ -34,29 +38,46 @@ library KeepFactorySelection {
 
         KeepFactorySelector factorySelector;
 
-        // Standard ECDSA keep factory: KEEP stake and ETH bond.
+        // Standard ECDSA keep vendor and factory: KEEP stake and ETH bond.
         // Guaranteed to be set for initialized factory.
+        IBondedECDSAKeepVendor keepStakeVendor;
         IBondedECDSAKeepFactory keepStakeFactory;
 
-        // Fully backed ECDSA keep factory: ETH stake and ETH bond.
-        IBondedECDSAKeepFactory ethStakeFactory;
+        // Fully backed ECDSA keep vendor and factory: ETH bond only.
+        IBondedECDSAKeepVendor fullyBackedVendor;
+        IBondedECDSAKeepFactory fullyBackedFactory;
+
+        // Lock for factories versions freeze. When set to true vendor won't be
+        // called to obtain a new factory address but a version of a factory
+        // from the moment of freezing will be used. Once locked the factory
```

Comment:
I believe we should have a coma after *Once locked*.

Reply:
b35547358432c1031368e630f51602fd1146d732